Infrequent failing `test_download_dataset` test

Motivation
----------
The `test_download_dataset` is failing sometimes with the error:
```
[gw1] linux -- Python 3.9.9 /usr/bin/python
test_bin_rucio.py:1048: in test_download_dataset
    assert re.search(search, err) is not None
E   AssertionError: assert None is not None
E    +  where None = <function search at 0x7f252a347ca0>('file_OQAYTFTGUO successfully downloaded',
```

As @yuyiguo pointed out in https://github.com/rucio/rucio/pull/5872#issuecomment-1252920038, the test does not download the dataset because it is already available locally. There seems to be no pattern on when the test is failing and when not.

```
2022-09-19 22:32:02,995	INFO	Preparing download of data13_hip:file_OQAYTFTGUO
2022-09-19 22:32:02,995	INFO	File exists already locally: data13_hip:file_OQAYTFTGUO
```

Modification
------------
Remove the file before the download, so it does not exist locally and thus needs to be downloaded.

--- a/lib/rucio/tests/test_bin_rucio.py
+++ b/lib/rucio/tests/test_bin_rucio.py
@@ -1039,6 +1039,9 @@ class TestBinRucio:
         print(self.marker + cmd)
         exitcode, out, err = execute(cmd)
         print(out, err)
+
+        os.remove(tmp_file1)
+
         # download dataset
         cmd = 'rucio -v download --dir /tmp {0}'.format(tmp_dataset)  # triming '/tmp/' from filename
         print(self.marker + cmd)

